src/main/java/edu/kis/powp/jobs2d/drivers/decorators/subscribers/UsageMonitorDecoratorSubscriber.java

```diff
@@ -0,0 +1,22 @@
+package edu.kis.powp.jobs2d.drivers.decorators.subscribers;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.drivers.decorators.Job2dDriverUsageMonitorDecorator;
+import edu.kis.powp.jobs2d.features.DriverFeature;
+import edu.kis.powp.observer.Subscriber;
+
+public class UsageMonitorDecoratorSubscriber implements Subscriber {
+	private boolean useMonitor = false;
+
+	public void changeMonitoringState() {
+		useMonitor = !useMonitor;
+	}
+
+	@Override
+	public void update() {
+		Job2dDriver currentDriver = DriverFeature.getDriverManager().getCurrentDriver();
+		if (!currentDriver.getClass().equals(Job2dDriverUsageMonitorDecorator.class) && useMonitor) {
```

Comment:
it would be nice if checking via class type could be avoided (maybe a flag would be sufficient).